### ISSUE TITLE
v1.50.3 the HBP Services are down and Nextcloud Check shows error

### ISSUE THREAD

**Issue by fabiangerhard** [nextcloud/nextcloudpi#1664]
Hey, after updating to NextCloudPi version  v1.50.3 the HBP Services are down and Nextcloud Check shows error. 

**System information**

<--! Paste this in GitHub report -->

<details>
<summary>NextCloudPi diagnostics</summary>

```

NextCloudPi version  v1.50.3
NextCloudPi image    NextCloudPi_06-09-22
OS                   Debian GNU/Linux 11. 5.15.32-v8+ (aarch64)
automount            yes
USB devices          sda sdb 
datadir              /media/myCloudDrive/ncdata/data
data in SD           no
data filesystem      btrfs
data disk usage      692G/1.9T
rootfs usage         25G/117G
swapfile             /var/swap
dbdir                /var/lib/mysql
Nextcloud check      error
HTTPD service        up
PHP service          up
MariaDB service      up
Redis service        up
HPB service          down
Postfix service      up
Internet check       ok
public IP            ***REMOVED SENSITIVE VALUE***
public IP            ***REMOVED SENSITIVE VALUE***
Port check 80        closed
Port check 443       closed
IP                   ***REMOVED SENSITIVE VALUE***
gateway              ***REMOVED SENSITIVE VALUE***
Interface            eth0
certificates         ***REMOVED SENSITIVE VALUE***
NAT loopback         yes
Uptime               5days

```
</details>

<details>
<summary>Nextcloud configuration</summary>

```

An unhandled exception has been thrown:
Doctrine\DBAL\Exception: Failed to connect to the database: An exception occurred in the driver: could not find driver in /var/www/nextcloud/lib/private/DB/Connection.php:139
Stack trace:
#0 /var/www/nextcloud/3rdparty/doctrine/dbal/src/Connection.php(1519): OC\DB\Connection->connect()
#1 /var/www/nextcloud/3rdparty/doctrine/dbal/src/Connection.php(1041): Doctrine\DBAL\Connection->getWrappedConnection()
#2 /var/www/nextcloud/lib/private/DB/Connection.php(261): Doctrine\DBAL\Connection->executeQuery()
#3 /var/www/nextcloud/3rdparty/doctrine/dbal/src/Query/QueryBuilder.php(345): OC\DB\Connection->executeQuery()
#4 /var/www/nextcloud/lib/private/DB/QueryBuilder/QueryBuilder.php(281): Doctrine\DBAL\Query\QueryBuilder->execute()
#5 /var/www/nextcloud/lib/private/AppConfig.php(411): OC\DB\QueryBuilder\QueryBuilder->execute()
#6 /var/www/nextcloud/lib/private/AppConfig.php(176): OC\AppConfig->loadConfigValues()
#7 /var/www/nextcloud/lib/private/AppConfig.php(367): OC\AppConfig->getApps()
#8 /var/www/nextcloud/lib/private/legacy/OC_App.php(983): OC\AppConfig->getValues()
#9 /var/www/nextcloud/lib/private/Server.php(713): OC_App::getAppVersions()
#10 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(162): OC\Server->OC\{closure}()
#11 /var/www/nextcloud/3rdparty/pimple/pimple/src/Pimple/Container.php(122): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}()
#12 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(129): Pimple\Container->offsetGet()
#13 /var/www/nextcloud/lib/private/ServerContainer.php(136): OC\AppFramework\Utility\SimpleContainer->query()
#14 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(57): OC\ServerContainer->query()
#15 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(184): OC\AppFramework\Utility\SimpleContainer->get()
#16 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(162): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}()
#17 /var/www/nextcloud/3rdparty/pimple/pimple/src/Pimple/Container.php(118): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}()
#18 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(129): Pimple\Container->offsetGet()
#19 /var/www/nextcloud/lib/private/ServerContainer.php(136): OC\AppFramework\Utility\SimpleContainer->query()
#20 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(57): OC\ServerContainer->query()
#21 /var/www/nextcloud/lib/private/Server.php(1105): OC\AppFramework\Utility\SimpleContainer->get()
#22 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(162): OC\Server->OC\{closure}()
#23 /var/www/nextcloud/3rdparty/pimple/pimple/src/Pimple/Container.php(122): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}()
#24 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(129): Pimple\Container->offsetGet()
#25 /var/www/nextcloud/lib/private/ServerContainer.php(136): OC\AppFramework\Utility\SimpleContainer->query()
#26 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(57): OC\ServerContainer->query()
#27 /var/www/nextcloud/lib/private/Server.php(2082): OC\AppFramework\Utility\SimpleContainer->get()
#28 /var/www/nextcloud/lib/private/Files/View.php(122): OC\Server->getLockingProvider()
#29 /var/www/nextcloud/lib/private/Server.php(454): OC\Files\View->__construct()
#30 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(162): OC\Server->OC\{closure}()
#31 /var/www/nextcloud/3rdparty/pimple/pimple/src/Pimple/Container.php(122): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}()
#32 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(129): Pimple\Container->offsetGet()
#33 /var/www/nextcloud/lib/private/ServerContainer.php(136): OC\AppFramework\Utility\SimpleContainer->query()
#34 /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(57): OC\ServerContainer->query()
#35 /var/www/nextcloud/lib/private/Server.php(1445): OC\AppFramework\Utility\SimpleContainer->get()
#36 /var/www/nextcloud/lib/base.php(602): OC\Server->boot()
#37 /var/www/nextcloud/lib/base.php(1111): OC::init()
#38 /var/www/nextcloud/console.php(48): require_once('...')
#39 /var/www/nextcloud/occ(11): require_once('...')
#40 {main}
```
</details>

<details>
<summary>HTTPd logs</summary>

```

[Wed Nov 23 20:04:01.608084 2022] [authz_host:error] [pid 1005741:tid 547973976448] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to /img/info-white.svg failed, reason: unable to get the remote host name
[Wed Nov 23 20:04:01.666202 2022] [authz_host:error] [pid 1005741:tid 547973976448] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to /img/files.svg failed, reason: unable to get the remote host name
[Wed Nov 23 20:04:01.928262 2022] [authz_host:error] [pid 1005741:tid 547973976448] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to /ncp-launcher.php failed, reason: unable to get the remote host name
[Wed Nov 23 20:04:01.930323 2022] [authz_host:error] [pid 1005741:tid 547982377344] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to /img/favicon.png failed, reason: unable to get the remote host name
[Wed Nov 23 20:04:01.930323 2022] [authz_host:error] [pid 1005741:tid 547948773760] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to /ncp-output.php failed, reason: unable to get the remote host name
[Wed Nov 23 20:04:11.804183 2022] [authz_host:error] [pid 1005741:tid 547973976448] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to /ncp-launcher.php failed, reason: unable to get the remote host name
[Wed Nov 23 20:04:11.920285 2022] [authz_host:error] [pid 1005741:tid 547973976448] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to /ncp-launcher.php failed, reason: unable to get the remote host name
[Wed Nov 23 20:04:11.920311 2022] [authz_host:error] [pid 1005741:tid 547982377344] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to /img/download.svg failed, reason: unable to get the remote host name
[Wed Nov 23 20:04:11.920521 2022] [authz_host:error] [pid 1005741:tid 547957174656] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to /img/delete.svg failed, reason: unable to get the remote host name
[Wed Nov 23 20:04:18.744824 2022] [authz_host:error] [pid 1005741:tid 547957174656] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to /img/info.svg failed, reason: unable to get the remote host name
[Wed Nov 23 20:04:29.753089 2022] [authz_host:error] [pid 1005741:tid 547957174656] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to /ncp-launcher.php failed, reason: unable to get the remote host name
[Wed Nov 23 20:04:57.640974 2022] [authz_host:error] [pid 1005741:tid 547957174656] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to /img/red-circle.svg failed, reason: unable to get the remote host name
[Wed Nov 23 20:11:10.589051 2022] [authz_host:error] [pid 1005741:tid 547957174656] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to / failed, reason: unable to get the remote host name
[Wed Nov 23 20:11:10.760135 2022] [authz_host:error] [pid 1005741:tid 547957174656] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to / failed, reason: unable to get the remote host name
[Wed Nov 23 20:11:10.760448 2022] [authz_host:error] [pid 1005741:tid 547957174656] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to /index.php failed, reason: unable to get the remote host name
[Wed Nov 23 20:11:10.929971 2022] [authz_host:error] [pid 1005741:tid 547957174656] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to /index.php failed, reason: unable to get the remote host name
[Wed Nov 23 20:11:12.467428 2022] [authz_host:error] [pid 1005741:tid 547957174656] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to /ncp-launcher.php failed, reason: unable to get the remote host name
[Wed Nov 23 20:11:12.468592 2022] [authz_host:error] [pid 1005741:tid 547948773760] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to /ncp-output.php failed, reason: unable to get the remote host name
[Wed Nov 23 20:11:22.593318 2022] [authz_host:error] [pid 1005741:tid 547957174656] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to /ncp-launcher.php failed, reason: unable to get the remote host name
[Wed Nov 23 20:11:22.621880 2022] [authz_host:error] [pid 1005741:tid 547957174656] [client 192.168.178.38:61480] AH01753: access check of 'localhost' to /ncp-launcher.php failed, reason: unable to get the remote host name

```
</details>

<details>
<summary>Database logs</summary>

```

tail: cannot open '/var/log/mysql/*.log' for reading: No such file or directory

```
</details>
![Pasted Graphic 13](https://user-images.githubusercontent.com/71646855/203633070-7980ded7-8a38-4796-911d-71ba91bbe237.png)

**Comment by rustyb78** [comment]
I have the same issue.

**Comment by rustyb78** [comment]
> APT upgrade solves it

It doesn't fix mine at all, but, my instance is functioning despite these issues.

**Comment by Electrotec81** [comment]
I had the same problem when I ran a package update. This resulted in "php 8.2.0 CLI" being installed.
Before the update, everything worked fine with "php 8.1.3 CLI). Now both versions are installed.
However, Nextcloud now always uses the newer version, which leads to the error described. As a result, cron jobs are no longer executed for me and the notify_push service is not active. When I run a cron job manually, I get:
"This version of Nextcloud is not compatible with PHP>=8.2.<br/>You are currently running 8.2.0"

Maybe it's the same with you. Then you could possibly downgrade PHP to 8.1.3. Unfortunately, I'm not sure how to safely do this without risking errors.


NC: 24.0.5.1
NCP: 1.50.3

**Comment by chribre** [comment]
Hey everyone, 
 I'm not sure if it helps but there's another [thread on nextcloud forum](https://help.nextcloud.com/t/hold-on-php-8-1-after-php8-2-upgrade-yesterday/151893)  which more or less explains the circumstances. 
 
> I don’t use NCP, but when I issue apt install php-json on my Debian 11 instance, on which I have the sury repos enabled,     it tries to install multiple PHP8.2 packages...

 In my case using `sudo update-alternatives --config php` did the trick.

**Comment by fabiangerhard** [comment]
@chribre thanks, i did a clean install of the image and now it works.

**Comment by ZendaiOwl** [comment]
We'll need to implement a package hold on the PHP packages and see if that stops the automatic updates and upgrades from installing the newer versions of PHP, I think the `sudo apt-mark hold {package-name}` should work for this

https://www.cyberciti.biz/faq/apt-get-hold-back-packages-command/

**Comment by OfficialMuffin** [comment]
My `ncp-report` also showed that `HPB Service` was also down.
When I did:

```
sudo systemctl reload apache2.service
sudo systemctl restart apache2.service
```

My problem was resolved. `HPB Service` was now showing up

**Comment by theCalcaholic** [comment]
Should be fixed in v1.51.0
